htdocs/userinfo.php

```diff
@@ -39,7 +39,7 @@
 
 $xoopsOption['pagetype'] = 'user';
 include 'mainfile.php';
-$hash_uid = (int) $_GET['uid'];
+$hash_uid = $_GET['uid'];
 
```

Comment:
The input should be filtered and tested before using. In the original code, we force it to be an integer before proceeding

Reply:
This is a little better. I recommend reviewing the icms_core_DataFilter::checkVar() method. Strings can still be malicious and validating input and access are the best defense against breaches.